src/core/resolvers/require.zig

```diff
@@ -142,25 +176,61 @@ pub fn zune_require(L: *VM.lua.State) !i32 {
     };
     defer if (opened_dir) dir.close();
 
-    var resolvedPath: ?[]const u8 = null;
-    defer if (resolvedPath) |path| allocator.free(path);
+    var resolvedPath: ?[]u8 = null;
```

Comment:
Must manage near, as cases anywhere in between this variable and the next `defer` for this, would cause a leak. 
```suggestion
    var resolvedPath: ?[]u8 = null;
    defer if (resolvedPath) |r| allocator.free(r);
```